models/gans/wgan.py

```diff
@@ -1,12 +1,181 @@
 # TODO: Implement the wgan architecture (issue #6)
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.autograd import Variable
+from torch import autograd
+import time as t
+import matplotlib.pyplot as plt
+plt.switch_backend('agg')
+import os
+from itertools import chain
+from torchvision import utils as utils
+from utils.logging import logging
 
-class WGAN:
-    pass
+SAVE_PER_TIMES = 100
```

Comment:
I think we should put the ref of the code on the first line: 
https://github.com/Zeleni9/pytorch-wgan/blob/master/models/wgan_gradient_penalty.py